### PR TITLE
MAINT-37593 : handle AppCenter NPE with lost apps images

### PR DESCRIPTION
**Pull request by hakermi**
This PR should handle the NPE when the app image is lost 
and assign the default image when the app image is missing.

### CODE REVIEW

File: app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java [hunk 1]
```diff
@@ -25,6 +26,7 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import jdk.internal.org.jline.utils.Log;
```
Comment (rdenarie) [inline app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java]: unecessary import
```suggestion
```

File: app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java [hunk 2]
```diff
@@ -303,8 +305,16 @@ private Application toDTO(ApplicationEntity applicationEntity) throws FileStorag
...
+    String imageFileName;
+    FileItem file;
+    if(applicationEntity.getImageFileId()>0){
+      file =fileService.getFile(applicationEntity.getImageFileId());
+      if(file != null){
+        imageFileName = file.getFileInfo().getName();
+      }else {
+        imageFileName = null;
+      }
+    }
```
Comment (rdenarie) [inline app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java]:
```suggestion
    String imageFileName=null;
    if(applicationEntity.getImageFileId() != null && applicationEntity.getImageFileId()>0){
      File file =fileService.getFile(applicationEntity.getImageFileId());
      if(file != null){
        imageFileName = file.getFileInfo().getName();
      }
    }
```

File: app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java [hunk 3]
```diff
@@ -330,8 +340,16 @@ private UserApplication toUserApplicationDTO(ApplicationEntity applicationEntity
...
```
Comment (rdenarie) [inline app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java]: String imageFileName=null;
    if(applicationEntity.getImageFileId() != null && applicationEntity.getImageFileId()>0){
      File file =fileService.getFile(applicationEntity.getImageFileId());
      if(file != null){
        imageFileName = file.getFileInfo().getName();
      }
    }

File: app-center-webapps/src/main/webapp/vue-apps/appLauncher/components/AppLauncherDrawer.vue [hunk 4]
```diff
@@ -73,7 +73,7 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
                   :href="application.computedUrl"
                   @click="logOpenApplication(application.id)"
                 >
-                  <img v-if="application.imageFileId" class="appLauncherImage" :src="`/portal/rest/app-center/applications/illustration/${application.id}`" />
+                  <img v-if="application.imageFileId && application.imageFileName!==null" class="appLauncherImage" :src="`/portal/rest/app-center/applications/illustration/${application.id}`" />
```
Comment (ahamdi) [inline app-center-webapps/src/main/webapp/vue-apps/appLauncher/components/AppLauncherDrawer.vue]:
```suggestion
                  <img v-if="application.imageFileId && application.imageFileName" class="appLauncherImage" :src="`/portal/rest/app-center/applications/illustration/${application.id}`" />
```

File: app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java [hunk 5]
```diff
@@ -303,8 +308,17 @@ private Application toDTO(ApplicationEntity applicationEntity) throws FileStorag
...
       return null;
     }
-    String imageFileName = applicationEntity.getImageFileId() != null
-        && applicationEntity.getImageFileId() > 0 ? fileService.getFile(applicationEntity.getImageFileId()).getFileInfo().getName() : null;
+    String imageFileName = null;
+    FileItem file = null;
+    if(applicationEntity.getImageFileId()>0){
+      try {
+        file = fileService.getFile(applicationEntity.getImageFileId());
+        imageFileName = file.getFileInfo().getName();
```
Comment (ahamdi) [inline app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java]: You can use FileService.getFileInfo(fileId) WDYT ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: app-center-services/src/test/java/org/exoplatform/appcenter/service/ApplicationCenterServiceTest.java [hunk 6]
```diff
@@ -959,7 +959,7 @@ public void testGetImageStream() throws Exception {
     assertTrue(inputStream.available() > 0);
   }
 
-  @Test
+  @Test(expected=NullPointerException.class)
```
Comment (rdenarie) [inline app-center-services/src/test/java/org/exoplatform/appcenter/service/ApplicationCenterServiceTest.java]:
```suggestion
  @Test
```

File: app-center-services/src/test/java/org/exoplatform/appcenter/service/ApplicationCenterServiceTest.java [hunk 7]
```diff
@@ -1131,7 +1131,7 @@ public void testAddApplicationPlugin() throws FileStorageException {
     assertEquals(0, applicationsList.getApplications().size());
   }
 
-  @Test
+  @Test(expected=NullPointerException.class)
```
Comment (rdenarie) [inline app-center-services/src/test/java/org/exoplatform/appcenter/service/ApplicationCenterServiceTest.java]:
```suggestion
  @Test
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java (hunk 5 above)
Comment (hakermi) [reply app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java]: done

**Review by ahamdi — DISMISSED** [review]

File: app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java [hunk 8]
```diff
@@ -303,8 +306,14 @@ private Application toDTO(ApplicationEntity applicationEntity) throws FileStorag
...
-    String imageFileName = applicationEntity.getImageFileId() != null
-        && applicationEntity.getImageFileId() > 0 ? fileService.getFile(applicationEntity.getImageFileId()).getFileInfo().getName() : null;
+    String imageFileName = null;
+    FileInfo fileInfo = null;
+    if(applicationEntity.getImageFileId() != null && applicationEntity.getImageFileId()>0){
+      fileInfo = fileService.getFileInfo(applicationEntity.getImageFileId());
+      if(fileInfo != null){
+        imageFileName = fileInfo.getName();
+      }
+    }
```
Comment (rdenarie) [inline app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java]:
```suggestion
    if(applicationEntity.getImageFileId() != null && applicationEntity.getImageFileId()>0){
      FileInfo fileInfo = fileService.getFileInfo(applicationEntity.getImageFileId());
      if(fileInfo != null){
        imageFileName = fileInfo.getName();
      }
    }
```

File: app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java [hunk 9]
```diff
@@ -330,8 +339,14 @@ private UserApplication toUserApplicationDTO(ApplicationEntity applicationEntity
...
```
Comment (rdenarie) [inline app-center-services/src/main/java/org/exoplatform/appcenter/storage/ApplicationCenterStorage.java]:
```suggestion
    String imageFileName = null;
    if(applicationEntity.getImageFileId() != null && applicationEntity.getImageFileId()>0){
      FileInfo fileInfo = fileService.getFileInfo(applicationEntity.getImageFileId());
      if(fileInfo != null){
        imageFileName = fileInfo.getName();
      }
    }
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: app-center-webapps/src/main/webapp/vue-apps/userSetup/components/UserFavoriteApplications.vue [hunk 10]
```diff
@@ -50,7 +50,7 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
         <v-list-item>
           <div class="favoriteAppImage">
             <a :target="favoriteApp.target" :href="favoriteApp.computedUrl" @click="logOpenApplication(favoriteApp.id)">
-              <img v-if="favoriteApp.imageFileId" class="appImage" :src="`/portal/rest/app-center/applications/illustration/${favoriteApp.id}`" />
+              <img v-if="favoriteApp.imageFileId && authorizedApp.imageFileName" class="appImage" :src="`/portal/rest/app-center/applications/illustration/${favoriteApp.id}`" />
```
Comment (rdenarie) [inline app-center-webapps/src/main/webapp/vue-apps/userSetup/components/UserFavoriteApplications.vue]:
```suggestion
              <img v-if="favoriteApp.imageFileId && favoriteApp.imageFileName" class="appImage" :src="`/portal/rest/app-center/applications/illustration/${favoriteApp.id}`" />
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**